### DISCUSSION TITLE
Error: No filings available.

### DISCUSSION THREAD

**Discussion by victoriaplummer** [sec-edgar/sec-edgar#245] [Q&A]
Hello,

I'm just getting started, and was trying to run the example. I wasn't able to run the current version without getting an error that the FilingType module wasn't found. Currently, I'm using the alpha, and when trying to run the example from the ReadMe I'm receiving the following error. 

`secedgar.exceptions.NoFilingsError: No filings available.`

Could anyone weigh in on what could be happening here? Code for reference below:

```
from secedgar import filings, FilingType
from datetime import date


my_path = '~/Filings'
my_start_date = date(2020, 12, 10),
my_end_date = date(2021, 12, 31),


# 10Q filings for Apple (ticker "aapl")
my_filings = filings(
                    cik_lookup='aapl',
                    start_date = my_start_date,
                    end_date= my_end_date,
                    filing_type=FilingType.FILING_10Q,
                    count = 4, 
                     )
                     
my_filings.save(my_path)
```

**Comment by jackmoody11** [comment]
> Hello,
> 
> 
> 
> I'm just getting started, and was trying to run the example. I wasn't able to run the current version without getting an error that the FilingType module wasn't found. Currently, I'm using the alpha, and when trying to run the example from the ReadMe I'm receiving the following error. 
> 
> 
> 
> `secedgar.exceptions.NoFilingsError: No filings available.`
> 
> 
> 
> Could anyone weigh in on what could be happening here? Code for reference below:
> 
> 
> 
> ```
> 
> from secedgar import filings, FilingType
> 
> from datetime import date
> 
> 
> 
> 
> 
> my_path = '~/Filings'
> 
> my_start_date = date(2020, 12, 10),
> 
> my_end_date = date(2021, 12, 31),
> 
> 
> 
> 
> 
> # 10Q filings for Apple (ticker "aapl")
> 
> my_filings = filings(
> 
>                     cik_lookup='aapl',
> 
>                     start_date = my_start_date,
> 
>                     end_date= my_end_date,
> 
>                     filing_type=FilingType.FILING_10Q,
> 
>                     count = 4, 
> 
>                      )
> 
>                      
> 
> my_filings.save(my_path)
> 
> ```

Hi, Victoria. You need to specify your `user_agent`. This is the SEC's way of knowing who is requesting data. Add the `user_agent` keyword argument to the filings function call. For example,

```python

filings(..., user_agent="Victoria (your_email@example.com)")
```

**Comment by victoriaplummer** [comment]
Hey Jack,

Thanks for the response!  When added a user agent I was getting an error as well. I tried creating an object from NetworkClient(), and then adding, client=my_client - but I got this error when trying to create the NetworkClient object : NetworkClient.__init__() got multiple values for argument

What should I try next?

Here are the errors I've run into and how I got around them so far:

* CompanyFilings
Error: Module wasn't found
Workaround: Switched to filings
Notes: using CompanyFilings is still in the documentation

* FilingType
Error: module wasn't found
Workaround: used the 4.0 alpha

* user_agent
Error: filings() got an unexpected keyword argument 'user_agent',
Workaround ⚠️  (still getting errors) : used client instead and created an object from the NetworkClient class.

* 
client / NetworkClient

Error: NetworkClient.__init__() got multiple values for argument 'retry_count'
Workaround ⚠️: IDK

Here's my current code:

from secedgar import filings , FilingType

from secedgar *.* client import NetworkClient

from datetime import date

my_path = '~/Filings'

my_start_date = date ( 2020 , 12 , 10 ) ,

my_end_date = date ( 2021 , 12 , 31 ) ,

my_client = NetworkClient ( 'Victoria ( ***@***.*** )' , retry_count = 3 )

# 10Q filings for Apple (ticker "aapl")

my_filings = filings (

cik_lookup = 'aapl' ,

start_date = my_start_date ,

end_date = my_end_date ,

filing_type = FilingType *.* FILING_10Q ,

client = my_client ,

count = 4 ,

)

print ( dir ( my_filings ))

print ( my_filings *.* _get_urls_for_cik ( '320193' ))

*Victoria Plummer*

Integration Engineer

—

Check out our upcoming webinars! ( https://airtable.docsend.com/view/2vhdhdcx8677naxg )

**Comment by victoriaplummer** [comment]
NVM got it!

*Victoria Plummer*

Integration Engineer

—

Check out our upcoming webinars! ( https://airtable.docsend.com/view/2vhdhdcx8677naxg )

**Reply by jackmoody11** [reply]
What ended up being the issue?

**Comment by victoriaplummer** [comment]
my bad code. wasn't putting user_agent= in front of my user agent string.

everything's working great now - thank you!

*Victoria Plummer*

Integration Engineer

—

Check out our upcoming webinars! ( https://airtable.docsend.com/view/2vhdhdcx8677naxg )

**Reply by jackmoody11** [reply]
Glad you figured it out!
